### PR TITLE
expanded supported languages

### PR DESCRIPTION
**Pull request by Frijol**
Used to be like

screen shot 2016-09-01 at 1 52 18 pm

Now more like

screen shot 2016-09-01 at 1 52 35 pm

r? @rwaldron

### CODE REVIEW

**Comment by Frijol** [comment]
any interest in adding more information here about the binary module server?

File: API/Supported_Languages.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Supported languages
+
+## JavaScript/Node
+Tessel 2 has full support for JavaScript and Node.JS. The relevant repos can be found here:
```
Comment (tcr) [inline API/Supported_Languages.md]: "Node.js"

File: API/Supported_Languages.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+* [Firmware support for JavaScript/Node](https://github.com/tessel/t2-firmware)
+* [CLI support for JavaScript/Node](https://github.com/tessel/t2-cli/blob/master/lib/tessel/deployment/javascript.js)
+
+### Binary modules
+There is some support for modules with binary dependencies on Tessel 2.
```
Comment (tcr) [inline API/Supported_Languages.md]: "There is support for binary modules on Tessel 2 that we have precompiled."

File: API/Supported_Languages.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+The best way to find out whether the module you want is supported is to try it. If it's supported, it will work! If it is not supported, you will get an error message like this ([Error message source in CLI](https://github.com/tessel/t2-cli/blob/master/lib/tessel/deployment/javascript.js)):
```
Comment (tcr) [inline API/Supported_Languages.md]: "The best way to find out whether the module you want is available is to try deploying it."

File: API/Supported_Languages.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    1. The binary is platform specific and cannot be compiled for OpenWRT.
+    2. A pre-compiled binary has not yet been generated for this module.
+    3. The binary didn't compile correctly for the platform that you're developing on.
+        It's possible that the binary is Linux-only or even OpenWRT specific,
+        try npm installing with "--force" and rerun your deployment command.
+    Please file an issue at https://github.com/tessel/t2-cli/issues/new
+```
+
+## Rust (work in progress)
+The Tessel team is working toward first-class support for Rust. If you're interested in the state of that project, check out the [tessel-rust repo](https://github.com/tessel/rust-tessel).
```
Comment (tcr) [inline API/Supported_Languages.md]: I think the URL is wrong.

File: API/Supported_Languages.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+## How to add support for more languages
+Interested in adding support for a new language to Tessel 2's CLI? Here is a detailed blog post on the six components you will need: [Interfacing with the Language Plugin API for Tessel 2](https://tessel.io/blog/148706216397/interfacing-with-the-language-plugin-api-for)
+
+It is worth noting that [support of Python for Tessel 2](https://github.com/tcr/tessel-python) was originally planned, but is no longer in active development. [Read the blog post](https://tessel.io/blog/146714850172/ramping-up-rust-backing-away-from-python-johnny) explaining this decision.
```
Comment (tcr) [inline API/Supported_Languages.md]: Surround paragraph in parentheses

File: API/Supported_Languages.md (hunk 4 above)
Comment (Frijol) [inline API/Supported_Languages.md]: surprisingly, both are correct

**Pull request merged**